pkgs/top-level/stage.nix

```diff
@@ -117,6 +117,33 @@ let
     lib.optionalAttrs allowCustomOverrides
       ((config.packageOverrides or (super: {})) super);
 
+  # Override system. This is useful to build i686 packages on x86_64-linux.
+  forceSystem = system: kernel: nixpkgsFun {
+    localSystem = {
+      inherit system;
+      platform = stdenv.hostPlatform.platform // { kernelArch = kernel; };
+    };
+  };
+
+  # Convenience attributes for instantitating nixpkgs. Each of these
+  # will instantiate a new version of allPackages. They map example
+  # attributes to their own thing.
+  extraPkgs = self: super: {
+     pkgsCross = lib.mapAttrs (n: crossSystem:
+                              nixpkgsFun { inherit crossSystem; })
+                              lib.systems.examples;
+     pkgsLocal = lib.mapAttrs (n: localSystem:
+                              nixpkgsFun { inherit localSystem; })
+                              lib.systems.examples;
+
+     # Used by wine, firefox with debugging version of Flash, ...
+     pkgsi686Linux = forceSystem "i686-linux" "i386";
+
+     callPackage_i686 = if stdenv.system == "i686-linux" || stdenv.system == "x86_64-linux"
```

Comment:
`if stdenv.hostPlatform.isx86` I think we can simplify this now

Reply:
Okay to just remove the check? It should be done in another place.